bin/improver-wind-downscaling

```diff
@@ -113,6 +123,21 @@ def main():
     non_dim_coords = [x.name() for x in wind_speed.coords(dim_coords=False)]
     if 'realization' in non_dim_coords:
         wind_speed = iris.util.new_axis(wind_speed, 'realization')
+
+    if args.output_height_level:
+        constraints, units = parse_constraint_list(
+            ['height={}'.format(args.output_height_level[0])],
+            [args.output_height_level[1]])
+        single_level = extract_subcube(wind_speed, constraints, units)
```

Comment:
Does the wind downscaling "RoughnessCorrection" require the whole 3D wind field in order to correct each level?  If there is no interdependence between levels (which having looked at the plugin there doesn't seem to be - scaling parameters seem to be read from files rather than calculated on the fly), it would be more efficient to extract the wind on this level before performing the correction, so that we are not performing the downscaling on height levels that we don't actually need.

Reply:
OK, this is me not understanding the plugin correctly.  So this order of operations is fine.

Reply:
Multi-level wind data is needed for the downscaling, otherwise we would simply use the 10m wind speed diagnostic as input to the whole process.